src/concrete/ml/quantization/base_quantized_op.py

```diff
@@ -945,27 +950,6 @@ def cnp_round(
         Returns:
             numpy.ndarray: The rounded array.
         """
-
-        # Rounding is applied only if specified by user
-        if self.rounding_threshold_bits is not None:
-            if calibrate_rounding:
-                assert_true(
-                    not isinstance(x, fhe.tracing.Tracer),
-                    "Can't compute lsbs_to_remove at compilation time.",
-                )
-                assert_true(
-                    self.lsbs_to_remove is None,
-                    "Rounding has already been calibrated.",
-                )
-
-                current_n_bits_accumulator = compute_bits_precision(x)
-                self.lsbs_to_remove = current_n_bits_accumulator - self.rounding_threshold_bits
-
-            # mypy
-            assert self.lsbs_to_remove is not None
-
-            # Apply rounding if needed
-            if self.lsbs_to_remove > 0:
-                x = fhe.round_bit_pattern(x, lsbs_to_remove=self.lsbs_to_remove)
-
+        if self.rounder is not None and not calibrate_rounding:
+            x = fhe.round_bit_pattern(x, lsbs_to_remove=self.rounder)
```

Comment:
So this won't work because you have not compiled at this point. Compilation isn't part of the calibration it happens afterward.

Reply:
Actually this is not executed at calibration time so no failures.